flow/DeterministicRandom.cpp

```diff
@@ -93,11 +93,12 @@ uint64_t DeterministicRandom::randomUInt64() {
 }
 
 uint32_t DeterministicRandom::randomSkewedUInt32(uint32_t min, uint32_t maxPlusOne) {
-	std::uniform_real_distribution<double> distribution(std::log(min), std::log(maxPlusOne - 1));
-	double logpower = distribution(random);
-	uint32_t loguniform = static_cast<uint32_t>(std::pow(10, logpower));
-	// doubles can be imprecise, so let's make sure we don't violate an edge case.
-	return std::max(std::min(loguniform, maxPlusOne - 1), min);
+	ASSERT(min < maxPlusOne);
+	std::uniform_real_distribution<double> distribution(std::log(std::max<double>(min, 1.0 / M_E)),
```

Comment:
This breaks Windows build: `C:\ci\fdb_pr_builder\run\foundationdb\flow\DeterministicRandom.cpp(97,91): error : use of undeclared identifier 'M_E' [C:\ci\fdb_pr_builder\run\build\flow\flow.vcxproj]`

Reply:
Apparently Windows needs an extra `#define _USE_MATH_DEFINES` before including `<cmath>`.

https://docs.microsoft.com/en-us/cpp/c-runtime-library/math-constants?view=msvc-170